### ISSUE TITLE
Different behavior from Excel reader in different environments

### ISSUE THREAD

**Issue by kthyng** [intake/intake#822]
I'm getting different behavior for reading in excel files in two different environments and I wasn't able to figure out why. 

Here is simplified example code to demonstrate the issue, along with a sample file. This works in an environment with out of date packages – except intake which I have to install special from github, and doesn't work in my up-to-date environment. However, everything works fine in both for the equivalent CSV setup.

I tried matching the old version of pandas (1.4.3) and the old fsspec (2022.5.0), but wasn't sure what else might be relevant

[test.xlsx](https://github.com/user-attachments/files/15795244/test.xlsx)

```
file = "test.xlsx"

data = intake.readers.datatypes.Excel(file)
reader = data.to_reader("pandas:DataFrame")

reader.read()
```

In the version that works, a dataframe is returned, in the version that doesn't work, some weird bytes are returned, like this: b'PK\x03\x04\x14\x00\x06\x00\x08\x00\x00\x00!\x00b\xee\x9dh^\x01\x00\x00\x90\x04\x00\x00\x13\x00\x08\x02[Content_Types].xml \xa2\x04\x02

Any ideas? Thanks.

**Comment by martindurant** [comment]
Is the reader indeed the pandas-excel reader in each case? Getting the reader guessing right in every case has proved tricky and gone through a couple of iterations recently, that that might be what you are seeing. It still needs more work. You can always call `intake.readers.PandasExcel` directly. 

The bytes returned seem to be a ZIP archive, by the way https://en.wikipedia.org/wiki/ZIP_(file_format)#File_headers .

**Comment by kthyng** [comment]
I switched out that line to have the following:

```
file = "test.xlsx"

data = intake.readers.PandasExcel(file)
reader = data.to_reader("pandas:DataFrame")

reader.read()
```

and I'm getting the same results in my current up-to-date environment. I should note too that when I restart I rotate through different results: sometimes I get back `True`, sometimes the gibberish, sometimes an `OpenFiles` object, and sometimes and object error of some sort. This happened yesterday too.

**Comment by kthyng** [comment]
I am still stuck on this for my excel files. Are you able to get the code snippet above to work with the test excel file?

**Comment by martindurant** [comment]
> Are you able to get the code snippet above to work with the test excel file?

Yes, it works fine. Note that `data` is already a reader, where you probably wanted a intake.readers.datatypes.Excel (`data.data` correctly constructs it).

**Comment by kthyng** [comment]
I had a breakthrough. By changing the call to `PandasExcel` according to your last comment,

```
reader = intake.readers.PandasExcel(file)
reader.read()
```

I got this out for this first time:

`ModuleNotFoundError: No module named 'openpyxl'`

which turned out to fix the original problem! Weird that it didn't just say that in the first place. Now that I have installed that package, both my original code snippet with `intake.readers.datatypes.Excel` and calling `PandasExcel` directly work for me. Thanks for helping to solve the mystery.
